patches/minecraft/net/minecraft/client/gui/screens/controls/KeyBindsList.java.patch

```diff
@@ -22,11 +44,19 @@
              KeyBindsList.this.f_93386_.f_91066_.m_92159_(p_193916_, p_193916_.m_90861_());
              KeyBindsList.this.m_269130_();
           }).m_252987_(0, 0, 50, 20).m_252778_((p_253313_) -> {
-@@ -144,7 +_,7 @@
+@@ -143,8 +_,14 @@
+ 
        public void m_6311_(GuiGraphics p_281805_, int p_281298_, int p_282357_, int p_281373_, int p_283433_, int p_281932_, int p_282224_, int p_282053_, boolean p_282605_, float p_281432_) {
           int k = p_281373_ + 90 - KeyBindsList.this.f_193859_;
-          p_281805_.m_280614_(KeyBindsList.this.f_93386_.f_91062_, this.f_193911_, k, p_282357_ + p_281932_ / 2 - 9 / 2, 16777215, false);
+-         p_281805_.m_280614_(KeyBindsList.this.f_93386_.f_91062_, this.f_193911_, k, p_282357_ + p_281932_ / 2 - 9 / 2, 16777215, false);
 -         this.f_193913_.m_252865_(p_281373_ + 190);
++         // NeoForge: Trim strings that are too long, and show a tooltip if the mouse is over the trimmed string
++         List<net.minecraft.network.chat.FormattedText> lines = KeyBindsList.this.f_93386_.f_91062_.m_92865_().m_92414_(this.f_193911_, NAME_SPLIT_LENGTH, net.minecraft.network.chat.Style.f_131099_);
++         MutableComponent nameComponent = Component.m_237113_(lines.get(0).getString() + (lines.size() > 1 ? "..." : ""));
```

Comment:
This likely should be declared as a `Component` rather than a `MutableComponent` and I suspect it may make sense to have it be:
```java
Component nameComponent = lines.size() > 1 ? Component.literal(lines.get(0).getString() + "...") : this.name;
```
That way if it doesn't need to split then it isn't creating a new component